File: Algorithm/10주_완성_알고리즘_코딩테스트_기초/1.선형_알고리즘_기초/1I_선택정렬_구현하기/1I.py
from typing import List


def get_min_index(data: List[int], begin: int, end: int):
    index = begin  
    for i in range(begin + 1, end + 1):
        if data[i] < data[index]:
            index = i

    return index


def selection_sort(data: List[int]):
    n = len(data)
    for i in range(n - 1):
        min_index = get_min_index(data, i, n - 1)
        data[i], data[min_index] = data[min_index], data[i]


if __name__ == "__main__":
	n = int(input())
	data = [int(w) for w in input().split()]
	selection_sort(data)
	print(' '.join([str(w) for w in data]))

File: Algorithm/10-Week_Coding_Test/1.선형_알고리즘_기초/1I_선택정렬_구현하기/1I.js
const fs = require('fs');

// 입력 읽기
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const n = Number(input[0]);
const data = input[1].split(' ').map(Number);

function getMinIndex(data, begin, end) {
    let index = begin;
    for (let i = begin + 1; i < end; i++) {
        if (data[i] < data[index]) {
            index = i;
        }
    }
    return index;
}

function selectionSort(data) {
    for (let i = 0; i < data.length - 1; i++) {
        const minIndex = getMinIndex(data, i, data.length);
        [data[i], data[minIndex]] = [data[minIndex], data[i]]; // Swap
    }
}

selectionSort(data);

console.log(data.join(' '));
